pandas/_libs/index.pyx

```diff
@@ -270,7 +270,15 @@ cdef class IndexEngine:
             Py_ssize_t i, j, n, n_t, n_alloc
 
         self._ensure_mapping_populated()
-        values = np.array(self._get_index_values(), copy=False)
+        #  GH #35392  Index.get_indexer_non_unique misbehaves when index contains multiple nan
```

Comment:
nit-pick: people usually leave comments for TO DO's. git/blame keeps track of code changes

Reply:
gotcha, removed the comment